service/collector/etcdbackup.go

```diff
@@ -238,15 +238,59 @@ func (d *ETCDBackup) Describe(ch chan<- *prometheus.Desc) error {
 }
 
 func (d *ETCDBackup) getTenantClusterIDs() ([]string, error) {
+	crdClient := d.g8sClient
 	var ret []string
 
-	azureConfigList, err := d.g8sClient.ProviderV1alpha1().AzureConfigs(v1.NamespaceAll).List(v1.ListOptions{})
-	if err != nil {
-		return ret, microerror.Mask(err)
+	// AWS
+	{
+		crdList, err := crdClient.ProviderV1alpha1().AWSConfigs(v1.NamespaceAll).List(v1.ListOptions{})
+		if err == nil {
+			for _, awsConfig := range crdList.Items {
+				// Only backup cluster if it was not marked for delete.
+				if awsConfig.DeletionTimestamp == nil {
```

Comment:
The behavior is effectively the same, but I've seen open source controllers using

```suggestion
				if awsConfig.ObjectMeta.DeletionTimestamp.IsZero() {
```

Reply:
done